bgpd/bgp_zebra.c

```diff
@@ -829,6 +840,22 @@ bgp_zebra_announce (struct prefix *p, struct bgp_info *info, struct bgp *bgp, sa
 	  else if (info->peer->nexthop.ifp)
 	    ifindex = info->peer->nexthop.ifp->ifindex;
 	}
+
+      if (strcmp (ifindex2ifname (ifindex), "lo") == 0)
```

Comment:
will the name be in `info->peer->ifname`?

Reply:
ifname is equal to 0 in our case